operator/internal/manifests/var.go

```diff
@@ -567,11 +549,6 @@ func defaultNodeAffinity(enableNodeAffinity bool) *corev1.NodeAffinity {
 // defaultPodAntiAffinity for components in podAntiAffinityComponents will
 // configure pods, of a LokiStack, to preferably not run on the same node
 func defaultPodAntiAffinity(componentLabel, stackName string) *corev1.PodAntiAffinity {
```

Comment:
Sorry, can not comment on the correct lines:

---

(in `configureAffinity`, L514-516):

```go
	if cSpec.PodAntiAffinity != nil {
		affinity.PodAntiAffinity = cSpec.PodAntiAffinity
	}
```

Should we "merge" the default with the custom PodAntiAffinity or is overwriting with the custom one enough?

---

(in `configureAffinity`, L518-520):

```go
	if affinity.NodeAffinity == nil && affinity.PodAntiAffinity == nil {
		return nil
	}
```

If I am not mistaken, then we now _always_ have a PodAntiAffinity, so the function will never return `nil`. I think we do not need the if-block anymore.

Reply:
1. We should overwrite, that's enough IMHO.
2. You are right the if branch is not needed anymore.